drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi_common.c

```diff
@@ -57,22 +62,23 @@ int spi_mcux_configure(const struct device *dev, const struct spi_config *spi_cf
 		return ret;
 	}
 
-	base->CR |= LPSPI_CR_RST_MASK;
-	base->CR |= LPSPI_CR_RRF_MASK | LPSPI_CR_RTF_MASK;
-	base->CR = 0x00U;
-
 	if (data->ctx.config != NULL) {
 		/* Setting the baud rate in LPSPI_MasterInit requires module to be disabled. Only
 		 * disable if already configured, otherwise the clock is not enabled and the
 		 * CR register cannot be written.
 		 */
 		LPSPI_Enable(base, false);
+
 		while ((base->CR & LPSPI_CR_MEN_MASK) != 0U) {
 			/* Wait until LPSPI is disabled. Datasheet:
 			 * After writing 0, MEN (Module Enable) remains set until the LPSPI has
 			 * completed the current transfer and is idle.
 			 */
 		}
+
+		base->CR |= LPSPI_CR_RST_MASK;
+		base->CR |= LPSPI_CR_RRF_MASK | LPSPI_CR_RTF_MASK;
```

Comment:
I think you can remove these lines as they have already initialized in LPSPI_MasterInit

Reply:
actually, I just realized, they are still in the wrong place

Reply:
@Raymond0225  I updated the PR to do an alternative workaround but I still need to test on S32 hardware, so it will remain on draft still